### ISSUE TITLE
More circular statistics, and use those for Distributions of Longitude (and Angle?)

### ISSUE THREAD

**Issue by mhvk** [astropy/astropy#14427]
### What is the problem this feature will solve?

In discussions on fixing `Distribution` so that `LatitudeDistribution` and `LongitudeDistribution` can be created, @eerovaher [noted](https://github.com/astropy/astropy/pull/14421#issuecomment-1437422136) that the standard `pdf_mean`, `pdf_std`, etc., are not really useful for `Longitude`, given the wrapping. We probably should use those in `stats.circstat` instead (or at least give the option to), but those do not have `median`, `mad`, and various sortings (including histograms) are missing as well (and are obviously tricky!).

### Describe the desired outcome

Ideally we implement a full complement of circular statistics. In https://github.com/astropy/astropy/pull/14421#issuecomment-1438702994, @nstarman linked to an introduction of circular statistics that includes a median, https://insidebigdata.com/2021/02/12/circular-statistics-in-python-an-intuitive-intro/

**Comment by mhvk** [comment]
cc @larrybradley and @mirca

**Comment by pllim** [comment]
There is also this issue:

* #13749

**Comment by nstarman** [comment]
I agree https://github.com/astropy/astropy/issues/13749 is an ongoing issue! And it would be nice to upstream. However, I don't think this PR exacerbates this issue as it adds statistics which should appear in any stats library, so we would need to make sure we and scipy have median stats, regardless.
